File: web/src/api/marketData.ts
import apiClient from './index';
import { toCamelCase } from './utils';

export interface DailyBar {
  tradeDate: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
  amount: number | null;
}
export interface DailyBarsResponse {
  symbol: string;
  market: string;
  interval: '1d';
  adjustment: 'forward';
  source: string | null;
  items: DailyBar[];
}
export const marketDataApi = {
  async dailyBars(symbol: string, endDate?: string, startDate?: string, signal?: AbortSignal): Promise<DailyBarsResponse> {
    return toCamelCase((await apiClient.get(`/api/v1/market-data/daily-bars/${encodeURIComponent(symbol)}`, {
      params: { start_date: startDate, end_date: endDate }, signal,
    })).data);
  },
};

File: src/finance_analysis/interfaces/api/v1/schemas/market_data.py
"""Read-only forward-adjusted daily market data."""

from datetime import date
from typing import Literal

from pydantic import BaseModel


class DailyBarItem(BaseModel):
    trade_date: date
    open: float
    high: float
    low: float
    close: float
    volume: int
    amount: float | None = None


class DailyBarsResponse(BaseModel):
    symbol: str
    market: str
    interval: Literal["1d"] = "1d"
    adjustment: Literal["forward"] = "forward"
    source: str | None = None
    items: list[DailyBarItem]
